File: src/main/java/com/assets/board/entity/Position.java
package com.assets.board.entity;

import jakarta.persistence.*;
import lombok.Data;

@Data
@Entity
@Table(name = "positions")
public class Position {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "symbol")
    private String symbol;

    @Column(name = "description", length = 500)
    private String description;

    @Column(name = "quantity")
    private Integer quantity;

    @Column(name = "mark_price")
    private Double markPrice;

    @Column(name = "position_value")
    private Double positionValue;

    @Column(name = "open_price")
    private Double openPrice;

    @Column(name = "cost_basis_price")
    private Double costBasisPrice;

    @Column(name = "fifo_pnl_unrealized")
    private Double fifoPnlUnrealized;

    @Column(name = "holding_period_date_time")
    private String holdingPeriodDateTime;

    @Column(name = "weight")
    private Double weight;
}

File: src/main/java/com/assets/board/entity/Position.kt
package com.assets.board.entity

import jakarta.persistence.*
import lombok.Data

@Entity
@Table(name = "positions")
data class Position (
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    @Column(name = "symbol")
    var symbol: String? = null,

    @Column(name = "description", length = 500)
    var description: String? = null,

    @Column(name = "quantity")
    var quantity: Int? = null,

    @Column(name = "mark_price")
    var markPrice: Double? = null,

    @Column(name = "position_value")
    var positionValue: Double? = null,

    @Column(name = "open_price")
    var openPrice: Double? = null,

    @Column(name = "cost_basis_price")
    var costBasisPrice: Double? = null,

    @Column(name = "fifo_pnl_unrealized")
    var fifoPnlUnrealized: Double? = null,

    @Column(name = "holding_period_date_time")
    var holdingPeriodDateTime: String? = null,

    @Column(name = "weight")
    var weight: Double? = null
)
